src/plots/barPlot.ts

```diff
@@ -113,6 +120,56 @@ export class Bar<X, Y> extends XYPlot<X, Y> {
     return this;
   }
 
+  public barEnd(): Plots.TransformableAccessorScaleBinding<X, number>;
+  public barEnd(end: number | Accessor<number>): this;
+  public barEnd(end: X | Accessor<X>, scale: Scale<X, number>): this;
+  /**
+   * Gets/sets the accessor for the bar "end", which is used to compute the
+   * width of vertical bars or the height of horizontal bars.
+   *
+   * Normally, bar width is determined by how many bars can fit in the given
+   * space (minus some padding).
+   *
+   * You can override this behavior and manually set the start and end
+   * coordinates for each bar.
+   *
+   * For example, if the bar chart is "vertical" and this accessor is defined,
+   * then the width of a bar is calculated as `abs(barEnd() - x())`.
+   */
+  public barEnd(end?: number | Accessor<number> | X | Accessor<X>, scale?: Scale<X, number>): any {
+    if (end == null) {
+      return this._propertyBindings.get(Bar._BAR_END_KEY);
+    }
+
+    this._bindProperty(Bar._BAR_END_KEY, end, scale);
+    this._updateWidthAccesor();
+    this._updateValueScale();
+    this.render();
+    return this;
+  }
+
+  public barAlignment(): BarAlignment;
+  public barAlignment(align: BarAlignment): this;
+  /**
+   * Gets/Sets the bar alignment. Valid values are `"start"`, `"middle"`, and
+   * `"end"`, which determines the meaning of the accessor of the bar's scale
+   * coordinate (e.g. "x" for vertical bars).
+   *
+   * For example, a value of "start" means the x coordinate accessor sets the
+   * left hand side of the rect.
+   *
+   * The default value is "middle", which aligns to rect so that it centered on
+   * the x coordinate
+   */
+  public barAlignment(align?: BarAlignment): any {
```

Comment:
I guess that's my point - users can do nonsensical things with the API. We're telling them "here's a combination that exists, but it's actually a trap because you'd never want to do it". It'd be better to structure the API such that it never happens to begin with imo. But that does sound unreasonably hard to accomplish in this circumstance.